### DISCUSSION TITLE
Richtext / HTMl Editor Component

### DISCUSSION THREAD

**Discussion by madcoda9000** [microsoft/fluentui-blazor#738] [Ideas]
Hi,

currently I'm struggleing a bit implementing a richtext editor for blazor wasm that is integrating in fluentui well. So, that would be an awsome component for this library.

Best Regards
Sascha

**Comment by vnbaaij** [comment]
It is on the list but, to be frank, I don't expect to see one appearing soon. These things are not that easy to build.

**Comment by dvoituron** [comment]
What type of **rich text editor** are you thinking of (Markdown, HTML, RTF, ...)?

**Reply by madcoda9000** [reply]
Mainly Html. Currently i try to implement summernote as there is no blazor component vailable.

**Reply by dvoituron** [reply]
I agree with @vnbaaij that it would be very complex to create and integrate into this Fluent library. Because many processes need to be validated, like filtering the resulting HTML code to ensure, in particular, that it complies with accessibility rules. Or to customize formatting elements (bolding/italics or table formatting). All the while ensuring that the result is responsive and easy to use.

But I'm curious to see what can be done.

**Reply by madcoda9000** [reply]
yes, i've expected an answer like that :-). For sure, it will be a lot of work.

Using summernote it is looking like that:

![image](https://github.com/microsoft/fluentui-blazor/assets/25504739/e2cbc761-ac7a-429b-a4ab-f3ab0c47dec8)

**Comment by mguinness** [comment]
> Because many processes need to be validated, like filtering the resulting HTML code to ensure, in particular, that it complies with accessibility rules.

Maybe a rich text editor that outputs Markdown instead might be an alternative?  Something like [Wysimark](https://github.com/portive/wysimark)?

**Reply by vnbaaij** [reply]
We don't plan on including something like this, mainly because we do not want to include any external libraries in our library /package. With that we would make it harder for people to deviate from the choices we make.

**Reply by mguinness** [reply]
I wasn't necessarily advocating the use of that library, merely stating that using Markdown as an alternative to HTML (with the associated issues).

**Reply by vnbaaij** [reply]
I was also not meaning that specific library but for supporting Markdown we would **always** need **some** external library, because we are not going to (be able to) build that ourselves.

**Reply by mguinness** [reply]
Would that also exclude a library like [.NET Community Toolkit](https://github.com/CommunityToolkit/dotnet)?

@michael-hawker Do you have any plans for Markdown support?

**Reply by dvoituron** [reply]
> Would that also exclude a library like [.NET Community Toolkit](https://github.com/CommunityToolkit/dotnet)?

Yes, we prefer to be dependent on a minimum number of libraries and let developers add them if they wish. You could add these libs in your project, in addition of FluentUI Blazor.

**Reply by mguinness** [reply]
> Yes, we prefer to be dependent on a minimum number of libraries and let developers add them if they wish. You could add these libs in your project, in addition of FluentUI Blazor.

Do you mean like the [MarkdownSection](https://www.fluentui-blazor.net/Lab/MarkdownSection) component that uses the [Markdig](https://github.com/xoofx/markdig) library?

**Reply by vnbaaij** [reply]
Yes, you can use the `MarkdownSection` as an example and add that to your own project. It is indeed dependant on the Markdig library. We use it for the demo site, but nt in the packages we create.

**Comment by Nicolas3347** [comment]
I would also like to have a Richtext editor component. This is a must-have. Something like this and where you can also add your own button with your own action.
![Editor](https://github.com/user-attachments/assets/53d537de-4a9e-493a-be04-928dfbbb7b84)

**Reply by franklupo** [reply]
https://quilljs.com/ is a good solution to embed in your component

**Reply by vnbaaij** [reply]
Like Daniele says, you can use an (sny) external solution for this. This is not something that must be coming from us.

**Comment by AlexHedley** [comment]
You could use **Blazored** **TextEditor**

- https://github.com/Blazored/TextEditor
- https://www.nuget.org/packages/Blazored.TextEditor
